src/burnchains/burnchain.rs

```diff
@@ -491,6 +492,21 @@ impl Burnchain {
         )
     }
 
+    /// Is this block either the first block in a reward cycle or
+    ///  right before the reward phase starts? This is the mod 0 or mod 1
+    ///  block. Reward cycle start events (like auto-unlocks) process *after*
+    ///  the first reward block, so this function is used to determine when
+    ///  that has passed.
+    pub fn is_before_reward_cycle(
+        first_block_ht: u64,
+        burn_ht: u64,
+        reward_cycle_length: u64,
+    ) -> bool {
+        let effective_height = burn_ht - first_block_ht;
```

Comment:
Should this be an explicit `.checked_sub(..).expect(..)`?

Reply:
Sure